Deprecated numpy functions

Since [version 1.24](https://numpy.org/doc/stable/release/1.24.0-notes.html), numpy has deprecated the aliases `np.bool` and `np.int`, which are still used at numerous occasions in autorift, leaving the user with an AttributeError.
It can be easily fixed by replacing all occurrence of `np.bool` with `bool` and `np.int` with `int`.

Also `np.asscalar` has also been removed in [version 1.23](https://numpy.org/doc/stable/release/1.23.0-notes.html). I think it can be safely replaced by calling `int()` instead.

Hi @jhkennedy,

yes, I was thinking about submitting a PR, but I was working on the conda install directly, not on a clone of the repository.
I tried cloning the repository and running `python setup.py install` as recommended in the [installation notes](https://github.com/nasa-jpl/autoRIFT/blob/master/docs/install.md#62-standalone) but I get the following error:
"geo_autoRIFT/autoRIFT/bindings/autoriftcoremodule.cpp:43:10: fatal error: opencv2/imgproc/imgproc.hpp: No such file or directory"
(also the setup.py raises a DeprecationWarning).
Note that I'm installing directly in the conda environment that I created for installing autorift in the first place, and that already installed all dependencies, as I could not find an environment file in this repository.
Could you please indicate the way to go forward?
Thanks!

--- a/geo_autoRIFT/autoRIFT/autoRIFT.py
+++ b/geo_autoRIFT/autoRIFT/autoRIFT.py
@@ -1294,16 +1294,16 @@ def arImgDisp_u(
         #   Preparation for parallel
         in_shape = xGrid.shape
         I_shape = I1.shape
-        shape_prod = np.asscalar(np.prod(in_shape))
+        shape_prod = np.prod(in_shape).item()
 
         #        import pdb
         #        pdb.set_trace()
-        XI1 = mp.RawArray("b", np.asscalar(np.prod(I_shape)))
+        XI1 = mp.RawArray("b", np.prod(I_shape).item())
         XI1_np = np.frombuffer(XI1, dtype=np.uint8).reshape(I_shape)
         np.copyto(XI1_np, I1)
         del I1
 
-        XI2 = mp.RawArray("b", np.asscalar(np.prod(I_shape)))
+        XI2 = mp.RawArray("b", np.prod(I_shape).item())
         XI2_np = np.frombuffer(XI2, dtype=np.uint8).reshape(I_shape)
         np.copyto(XI2_np, I2)
         del I2
@@ -1572,16 +1572,16 @@ def arImgDisp_s(
         #   Preparation for parallel
         in_shape = xGrid.shape
         I_shape = I1.shape
-        shape_prod = np.asscalar(np.prod(in_shape))
+        shape_prod = np.prod(in_shape).item()
 
         #        import pdb
         #        pdb.set_trace()
-        XI1 = mp.RawArray("f", np.asscalar(np.prod(I_shape)))
+        XI1 = mp.RawArray("f", np.prod(I_shape).item())
         XI1_np = np.frombuffer(XI1, dtype=np.float32).reshape(I_shape)
         np.copyto(XI1_np, I1)
         del I1
 
-        XI2 = mp.RawArray("f", np.asscalar(np.prod(I_shape)))
+        XI2 = mp.RawArray("f", np.prod(I_shape).item())
         XI2_np = np.frombuffer(XI2, dtype=np.float32).reshape(I_shape)
         np.copyto(XI2_np, I2)
         del I2

